packages/runtime/datastore-definitions/src/channel.ts

```diff
@@ -195,6 +195,7 @@ export interface IChannelServices {
 
 /**
  * Definitions of a channel factory. Factories follow a common model but enable custom behavior.
+ * TODO: what are the semantics of this?
```

Comment:
The runtime must be able to produce "channels" of the correct type of in-memory object for the collaboration session (channels typically being DDSs).  So for instance if the collaboration includes a SharedMap, we need a factory that can produce a SharedMap object.  The runtime will consult with a registry of these factories during load and when receiving "attach" ops (which indicate a new instance of a channel being introduced to the collaboration session) to produce the appropriate in-memory object.